manifests/pipecd/values.yaml

```diff
@@ -114,5 +114,30 @@ monitoring:
 # All directives inside this section will be directly sent to the prometheus chart.
 # Head to the below link to see all available values.
 # https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml
-# TODO: Prepare the default prometheus.yml
-prometheus: {}
+prometheus:
+  pushgateway:
+    enabled: false
+  serverFiles:
+    prometheus.yml:
+      rule_files:
+        - /etc/config/recording_rules.yml
+        - /etc/config/alerting_rules.yml
+
+      scrape_configs:
+        - job_name: prometheus
+          static_configs:
+            - targets:
+                - localhost:9090
+
+        - job_name: pipecd-gateway
+          scrape_interval: 2m
+          metrics_path: /stats/prometheus
+          kubernetes_sd_configs:
+            - role: service
```

Comment:
Maybe we should use the `pod` or `endpoint` role for better data.

Reply:
Well, apparently I don't get them enough. Would you mind telling me a little bit about the difference between the [roles](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) in brief?

Reply:
While I haven't given it a shot to check exactly, I assume the exposed metrics is the same; seems like I was wrong.

Reply:
Yes. You can check the details [here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config) and we can talk more about that in the MTG today. But in short here are some highlights:
- the difference is the target list, targets for `service` role are just K8s Services, so our monitoring targets will just be `gateway` service (e.g. `pipecd-gateway.default.svc:9095`) 
- Prometheus will send HTTP requests to that service target to collect the metrics data and ONE of the pod of that service will be selected to handle that HTTP request. And then the data is not enough for our case.
- the metrics data will not contain the label data about pod, container...

Reply:
Thank you. Most of what I'm wondering has been solved. I assumed the replica of gateway pod is just one, but it's variable right.